### PR TITLE
ci: build macos arm64 packages

### PR DESCRIPTION
**Pull request by id**
Fixes [EMQX-8528](https://emqx.atlassian.net/browse/EMQX-8528)

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/` dir
- [ ] For EMQX 4.x: `appup` files updated (execute `scripts/update-appup.sh emqx`)
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

### CODE REVIEW

**Review by qzhuyan — DISMISSED** [review]

File: changes/v4.4.12-en.md [hunk 1]
```diff
@@ -6,6 +6,10 @@
 
 - Users can define the `externalTrafficPolicy` of service in EMQX Helm Chart [#9527](https://github.com/emqx/emqx/pull/9527).
 
+- Start releasing packages for Amazon Linux 2 [#9498](https://github.com/emqx/emqx/pull/9498).
+
+- Start building MacOS packages for Apple Silicon hadrdware [#9558](https://github.com/emqx/emqx/pull/9558).
```
Comment (zmstone) [inline changes/v4.4.12-en.md:11]: tr:
```
新增 Apple Silicon M1/M2 安装包发布 [#9558](https://github.com/emqx/emqx/pull/9558)。
```

Comment (zmstone) [inline changes/v4.4.12-en.md:11]: people usual search (grep) for M1/M2
```suggestion
- Start building MacOS packages for Apple Silicon M1/M2 [#9558](https://github.com/emqx/emqx/pull/9558).
```

Comment (id) [reply changes/v4.4.12-en.md]: @zmstone and for amazon linux? (forgot to add it last time)

**Comment by id** [comment]
@zmstone rebased

**Review by zmstone — APPROVED** [review]

**Pull request merged**